src/test/java/org/cdc/gov/sdp/aphl/AphlS3ComponentTest.java

```diff
@@ -0,0 +1,86 @@
+package org.cdc.gov.sdp.aphl;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.EndpointInject;
+import org.apache.camel.Produce;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.test.spring.CamelSpringJUnit4ClassRunner;
+import org.apache.camel.test.spring.CamelTestContextBootstrapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.BootstrapWith;
+import org.springframework.test.context.ContextConfiguration;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+
+@RunWith(CamelSpringJUnit4ClassRunner.class)
+@BootstrapWith(CamelTestContextBootstrapper.class)
+@ContextConfiguration(locations = { "/AphlS3ComponentTest-context.xml" })
+@PropertySource("/application.properties")
+public class AphlS3ComponentTest {
+
+	@Autowired
+	protected CamelContext camelContext;
+
+	@EndpointInject(uri = "mock:foo")
+	protected MockEndpoint foo;
+
+	@Produce(uri = "direct:start")
+	protected ProducerTemplate template;
+
+	@Test
+	@DirtiesContext
+	public void testAphlS3Producer() throws Exception {
+		final String accessKey = "accessKey1";
```

Comment:
This looks like it's testing the logic in the component as opposed to the component itself.  Can you update to invoke the camel route?